plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/cache/DeltaLakeCacheKeyProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+
+import java.util.Optional;
+
+public class DeltaLakeCacheKeyProvider
+        implements CacheKeyProvider
+{
+    /**
+     * Get the cache key of a TrinoInputFile. Returns Optional.empty() if the file is not cacheable.
+     */
+    @Override
+    public Optional<String> getCacheKey(TrinoInputFile delegate)
+    {
+        // TODO: Consider caching of the Parquet checkpoint files within _delta_log
+        if (!delegate.location().path().contains("/_delta_log/")) {
```

Comment:
Why do we skip caching this directory?

Reply:
The `_delta_log` directory contains the files `_last_checkpoint` and `_trino_meta/extended_stats.json`. These are not immutable, so are tricky to cache. Also the commit files of the form `000...123.json` might not be immutable on ABFS. The checkpoint files should be immutable when accessed by Trino. We decided to leave it as a future optimization in https://github.com/trinodb/trino/pull/18719#discussion_r1474342447